api/src/main/java/org/openmrs/util/LiquibaseVersionFinder.java

```diff
@@ -0,0 +1,148 @@
+
+	private static final String LIQUIBASE_UPDATE_FOLDER = "liquibase-updates";
+	private static final String LIQUIBASE_UPDATE_FILENAME = "liquibase-update-to-latest.xml";
+
+	private File snapshotFolder;
+	private File updateFolder;
+	
+	public LiquibaseVersionFinder() {
+		snapshotFolder = new File(
+			this.getClass()
+				.getClassLoader()
+				.getResource( LIQUIBASE_SNAPSHOTS_FOLDER )
+				.getFile()
+		);
+
+		updateFolder = new File(
+			this.getClass()
+				.getClassLoader()
+				.getResource( LIQUIBASE_UPDATE_FOLDER )
+				.getFile()
+		);
+	}
+
+	public Set<List<String>> getLiquibaseChangesetCombinations() {
+		Set<List<String>> changesetCombinations = new HashSet<>();
+
+		for ( String snapshotVersion : getLiquibaseSnapshotVersions() ) {
+			List<String> changesetFilenames = new ArrayList<>();
+			
+			changesetFilenames.addAll( getLiquibaseSnapshotFilenames( snapshotVersion ) );
+			changesetFilenames.addAll( getApplicableLiquibaseUpdateFileNames( snapshotVersion ) );
+			
+			changesetCombinations.add( changesetFilenames );
+		}
+		
+		return changesetCombinations;
+	}
+
+	public Optional<String> getLatestLiquibaseSchemaSnapshotFilename() {
+		Optional<String> snapshotVersion = getLatestLiquibaseSnapshotVersion();
+		if ( snapshotVersion.isPresent() ) {
+			return Optional.of( LIQUIBASE_SNAPSHOTS_FOLDER + "/" + snapshotVersion.get() + "/" + LIQUIBASE_SCHEMA_FILENAME );
+		}
+		return Optional.empty();
+	}
+
+	public Optional<String> getLatestLiquibaseCoreDataSnapshotFilename() {
+		Optional<String> snapshotVersion = getLatestLiquibaseSnapshotVersion();
+		if ( snapshotVersion.isPresent() ) {
+			return Optional.of( LIQUIBASE_SNAPSHOTS_FOLDER + "/" + snapshotVersion.get() + "/" + LIQUIBASE_CORE_DATA_FILENAME );
+		}
+		return Optional.empty();
+	}
+
+	public List<String> getApplicableLiquibaseUpdateFileNames( String currentVersion ) {
+		return getApplicableLiquibaseUpdateVersions( currentVersion )
+			.stream()
+			.map( version -> LIQUIBASE_UPDATE_FOLDER + "/" + version + "/" + LIQUIBASE_UPDATE_FILENAME )
+			.collect( Collectors.toList());
+	}
+
+	public Optional<String> getLatestLiquibaseSnapshotVersion() {
+		return getLiquibaseSnapshotVersions()
+			.stream()
+			.max( new VersionComparator() );
+	}
+	
+	List<String> getApplicableLiquibaseUpdateVersions( String currentVersion) {
+		if ( currentVersion.isEmpty() ) {
+			throw new IllegalArgumentException( "current version must not be empty" );
+		}
+		
+		VersionComparator versionComparator = new VersionComparator();
+		
+		return getLiquibaseUpdateVersions()
+			.stream()
+			.filter( updateVersion -> versionComparator.compare( updateVersion, currentVersion ) > 0 )
+			.sorted( versionComparator )
+			.collect( Collectors.toList() );
+	}
+
+	public List<String> getLiquibaseSnapshotFilenames( String version ) {
+		return Arrays.asList(
+			LIQUIBASE_SNAPSHOTS_FOLDER + "/" + version + "/" + LIQUIBASE_SCHEMA_FILENAME,
```

Comment:
Although sometimes mere `/` work out of the box, just use the `File.separator` to cover all edge cases @WolfSchlegel